### DISCUSSION TITLE
Path/Arcs Edge Bundling

### DISCUSSION THREAD

**Discussion by mbehrisch** [visgl/deck.gl#9320] [Ideas]
Dear all,

I‘d like to bounce off an idea and would like to ask for advice on how to start this project. 

We work with Deck.gl for map rendering. Overall, we are super happy with the functionality and performance, but being a vis researcher I would love to establish edge bundling (https://d3-graph-gallery.com/bundle.html) on our maps with many paths. 

Just for inspiration here an image showing flight paths:
![image](https://github.com/user-attachments/assets/1e24cb96-d747-410e-a36b-13dbe74c7fc8)

Now my unordered and unsorted thoughts:
- I would not know if I should start extending ArcLayers or PathLayers 
- I would like to use d3js as a start for an bundling algorithm implementation (https://observablehq.com/@d3/hierarchical-edge-bundling)
- shaders could be an option 

I am open for all suggestions and thought processes.

**Comment by ibgreen** [comment]
@mbehrisch Thanks for reaching out. This is a very interesting topic indeed. 

First, we do have some prior art in this space.
- https://flowmap.gl/ 
- https://graph.gl/gatsby/examples (graph.gl does use some of the d3 algorithms to layout graphs).

> I would not know if I should start extending ArcLayers or PathLayers

My first reaction is neither. Instead you would want to create a composite layer (`EdgeBundleLayer`?) that runs the bundling algorithm on its input data and then renders an ArcLayer/PathLayer etc on the bundled "table".
